lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -873,17 +873,13 @@ def add_path(self, path: str, *, recursive: bool = False) -> bool:
         Returns:
             True if path was added else False.
         """
-        rules_path = Path(path)
-        if not rules_path.exists:
-            logger.warning("Alert rules path %s does not exist", path)
-            return False
-
-        if rules_path.is_dir():
-            self.alert_groups.extend(self._from_dir(rules_path, recursive))
-        elif rules_path.is_file():
-            self.alert_groups.extend(self._from_file(rules_path.parent, rules_path))
-
-        return True
+        path = Path(path)  # type: Path
```

Comment:
```diff
-path = Path(path) # type: Path
+path = Path(str(path))
```

Reply:
I want to `git blame` you (since you did the original logic), but it's annotated in the signature. Is there a failure case in not casting it here?

Reply:
It's a py3.5 problem only. If you wan't to get rid of the type comment, convert to str.